contracts/preset/ERC721PsiMysteryBox.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: MIT
+/**
+  ______ _____   _____ ______ ___  __ _  _  _ 
+ |  ____|  __ \ / ____|____  |__ \/_ | || || |
+ | |__  | |__) | |        / /   ) || | \| |/ |
+ |  __| |  _  /| |       / /   / / | |\_   _/ 
+ | |____| | \ \| |____  / /   / /_ | |  | |   
+ |______|_|  \_\\_____|/_/   |____||_|  |_|   
+                                              
+                                            
+ */
+
+pragma solidity ^0.8.0;
+
+import "@chainlink/contracts/src/v0.8/interfaces/VRFCoordinatorV2Interface.sol";
+import "@chainlink/contracts/src/v0.8/VRFConsumerBaseV2.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+import "fpe-map/contracts/FPEMap.sol";
+
+import "../ERC721Psi.sol";
+
+abstract contract ERC721PsiMysteryBox is VRFConsumerBaseV2, ERC721Psi {
+    using Strings for uint256;
+
+    uint32 private constant _callbackGasLimit = 200000;
+
+    uint256 private seed;
+
+    event RandomnessRequest(uint256 requestId);
+
+
+    constructor() {
+        VRFConsumerBaseV2(_coordinator());
+    }
+
+    /**
+     * @dev The metadata URI before revealing.
+     */
+    function _unrevealedURI() internal pure virtual returns (string memory) {
+        return "";
```

Comment:
Can we make this function's modifier `view` as this way we can read from the state and we can set the max supply from the constructor?

Reply:
same as max Supply, you can do the same trick here.